### PR TITLE
docs: slim down readme

### PR DESCRIPTION
**Pull request by jsjoeio**
## Description

This PR refactors the `README.md` to be more slim and concise with the goal of making it less overwhelming when someone lands on the repo page. 

I also moved the Documentation Table of Contents to `./docs/index.md` and some getting started information to `./docs/walkthrough.md`. Open to suggestions!

🖼️ [Rendered](https://github.com/coder/coder/blob/jsjoeio/slim-readme/README.md)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,58 +8,30 @@ Discord"](https://img.shields.io/badge/join-us%20on%20Discord-gray.svg?longCache
...
-## Run Coder *now*
-
-```curl -L https://coder.com/install.sh | sh```
-
-## What Coder does
-Coder creates remote development machines so you can develop your code from anywhere. #coder
-
-> **Note**:
-> Coder is in an alpha state, but any serious bugs are P1 for us so [please report them](https://github.com/coder/coder/issues/new/choose).
+Coder creates remote development machines so your team can develop from anywhere.
```
Comment (jsjoeio) [inline README.md:11]: Keep it short and simple.

File: README.md [hunk 2]
```diff
@@ -8,58 +8,30 @@ Discord"](https://img.shields.io/badge/join-us%20on%20Discord-gray.svg?longCache
...
-- Onboard instantly then stay up to date continuously
-
 **Manage less**
 
 - Ensure your entire team is using the same tools and resources
   - Rollout critical updates to your developers with one command
 - Automatically shut down expensive cloud resources
 - Keep your source code and data behind your firewall
 
-## How it works
```
Comment (jsjoeio) [inline README.md:39]: Diff is hard to read. I just moved these down a bit.

File: README.md [hunk 3]
```diff
@@ -81,66 +53,11 @@ Once installed, you can run a temporary deployment in dev mode (all data is in-m
...
-  - [Editors and IDEs](./docs/workspaces.md#editors-and-ides)
-  - [Workspace lifecycle](./docs/workspaces.md#workspace-lifecycle)
-  - [Updating workspaces](./docs/workspaces.md#updating-workspaces)
-
-## Community
-
-Join the community on [Discord](https://discord.gg/coder) and [Twitter](https://twitter.com/coderhq) #coder!
-
-[Suggest improvements and report problems](https://github.com/coder/coder/issues/new/choose)
+Visit our docs [here](./docs/index.md).
```
Comment (jsjoeio) [inline README.md:60]: I called it `index.md` because that was what first came to mind. Open to other suggestions.

File: README.md [hunk 4]
```diff
@@ -155,7 +72,13 @@ Please file [an issue](https://github.com/coder/coder/issues/new) if any informa
 
 ---
 
-_As of 5/27/22_
+_Last updated: 5/27/22_
```
Comment (jsjoeio) [inline README.md:75]: Felt like this was easier to understand

File: docs/walkthrough.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+coder create --template="yourTemplate" <workspaceName>
+```
+
+Connect to your workspace via SSH:
+
+```bash
+coder ssh <workspaceName>
+```
+
+## Modifying templates
```
Comment (jsjoeio) [inline docs/walkthrough.md:28]: Didn't know where to put this so left it in walkthrough 🤷🏼‍♂️

Comment (bpmct) [reply docs/walkthrough.md:28]: Can we call it "quickstart"? I've seen that term used across different project docs pages

File: docs/index.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
+# Coder Documentation
+
+## Table of Contents
```
Comment (bpmct) [inline docs/index.md:3]: We should probably add "walkthrough" to the table of contents

File: README.md [hunk 7]
```diff
@@ -81,66 +53,11 @@ Once installed, you can run a temporary deployment in dev mode (all data is in-m
...
-
-You can edit the Terraform template using a sample template:
-
-```sh
-coder templates init
-cd gcp-linux/
-vim main.tf
-coder templates update gcp-linux
-```
+Use `coder --help` to get a complete list of flags and environment variables. For a full walkthrough, follow [this guide](./docs/walkthrough.md).
```
Comment (bpmct) [inline README.md:56]: Suggest changing to Use our [quickstart guide](./docs/walkthrough.md) for a full walkthrough

File: README.md [hunk 8]
```diff
@@ -155,7 +72,13 @@ Please file [an issue](https://github.com/coder/coder/issues/new) if any informa
 
 ---
 
-_As of 5/27/22_
+_Last updated: 5/27/22_
+
+## Community
```
Comment (bpmct) [inline README.md:77]:
```suggestion
## Support and Community
```

**Review by bpmct — APPROVED** [review]
All these changes look good to me, I added some suggestions.

File: docs/walkthrough.md (hunk 5 above)
Comment (jsjoeio) [reply docs/walkthrough.md:28]: Yes! I'll change that now.

File: docs/index.md (hunk 6 above)
Comment (jsjoeio) [reply docs/index.md:3]: Oh yeah, good call.

File: README.md [hunk 9]
```diff
@@ -8,58 +8,30 @@ Discord"](https://img.shields.io/badge/join-us%20on%20Discord-gray.svg?longCache
...
 
-You can use any Web IDE ([code-server](https://github.com/coder/code-server), [projector](https://github.com/JetBrains/projector-server), [Jupyter](https://jupyter.org/), etc.), [JetBrains Gateway](https://www.jetbrains.com/remote-development/gateway/), [VS Code Remote](https://code.visualstudio.com/docs/remote/ssh-tutorial) or even a file sync such as [mutagen](https://mutagen.io/).
+- Build and test faster
+  - Leveraging cloud CPUs, RAM, network speeds, etc.
+- Access your environment from any place on any client (even an iPad)
+- Onboard instantly then stay up to date continuously
 
-<p align="center">
-  <img src="./docs/images/ide-icons.svg" height=72>
-</p>
```
[the remark this answers is not in the document]
Comment (jsjoeio) [reply README.md:60]: I like the second idea!

[the remark this answers is not in the document]
Comment (jsjoeio) [reply README.md:60]: > we've received some early feedback that the IDE/cloud icons helped explain what Coder does

I agree with this, but it also makes me wonder, does our copy not convey this already? It probably should! aka the IDE/Cloud icons should be additive and shouldn't be necessary to explain Coder.

File: README.md (hunk 8 above)
Comment (jsjoeio) [reply README.md:77]: added "Community and Support"

File: README.md (hunk 7 above)
Comment (jsjoeio) [reply README.md:56]: I like that! changed

**Review by kylecarbs — APPROVED** [review]

**Pull request merged**